clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -25,25 +25,40 @@
 import java.util.List;
 
 public class MetadataRequest extends AbstractRequest {
-    
+
     private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.METADATA.id);
     private static final String TOPICS_KEY_NAME = "topics";
 
     private final List<String> topics;
 
+    /**
+     * Constructor to support asking for no topic metadata
+     */
+    public MetadataRequest() {
```

Comment:
Not sure if this is the best way to do this. I need a different constructor for requesting a `null` list because `null` matches both the list and struct constructors with the same "specificity".

I am open to ideas for a better way to support requesting no topic metadata.